xstream/sinks/mqtt_sink.go

```diff
@@ -56,6 +58,19 @@ func (ms *MQTTSink) Configure(ps map[string]interface{}) error {
 		}
 	}
 
+	var qos byte = 0
+	if qos, ok := ps["qos"]; ok {
+		if v, ok := qos.(byte); ok {
+			qos = v
+		} else if v, ok := qos.(string); ok {
+			if v == "1" {
+				qos = 1
+			} else if v == "2" {
+				qos = 2
+			}
+		}
+	}
+
```

Comment:
1. I think `qos` specified in rule should be `int` type, it should not be `string` or `byte`.
2. Check the value of `qos`, if it's not 0, 1 or 2, then it should return an error `not valid qos value %d`
3. Convert to `byte` type when setting `MQTTSink` struct instance.

Reply:
1.This is c.Publish method “Publish(topic string, qos byte, retained bool, payload interface{}) Token”

Reply:
> 1. I think `qos` specified in rule should be `int` type, it should not be `string` or `byte`.
> 2. Check the value of `qos`, if it's not 0, 1 or 2, then it should return an error `not valid qos value %d`
> 3. Convert to `byte` type when setting `MQTTSink` struct instance.

1.This is c.Publish method “Publish(topic string, qos byte, retained bool, payload interface{}) Token”

Reply:
Yes, but I don't think the type passed `Configure(ps map[string]interface{})` will be `byte`. It should be `int`.